mutation_compactor.hh

```diff
@@ -408,7 +408,8 @@ public:
 
         if (!cr.empty()) {
             partition_is_not_empty(consumer);
-            _stop = consumer.consume(std::move(cr), t, is_live);
+            auto stop = consumer.consume(std::move(cr), t, is_live);
+            _stop = stop_iteration(stop && !sstable_compaction());
```

Comment:
I think that instead of digging this hole even deeper, we should just fill it, i.e. instead of mimicking a bad example for consistency's sake, let's fix the bad behavior instead.
Can be a follow-up if not trivial and you don't want to pile onto this PR.